aspnetcore/mvc/models/file-uploads/samples/2.x/SampleApp/Utilities/FileHelpers.cs

```diff
@@ -0,0 +1,268 @@
+            { ".jpg", new byte[] { 0xFF, 0xD8, 0xFF, 0xE0, 0xFF, 0xD8, 0xFF, 0xE1, 0xFF, 0xD8, 0xFF, 0xE8 } },
+            { ".png", new byte[] { 0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A } },
+            { ".pdf", new byte[] { 0x25, 0x50, 0x44, 0x46 } },
+            { ".ppt", new byte[] { 0xD0, 0xCF, 0x11, 0xE0, 0xA1, 0xB1, 0x1A, 0xE1 } },
+            { ".pptx", new byte[] { 0x50, 0x4B, 0x03, 0x04 } },
+            { ".xls", new byte[] { 0xD0, 0xCF, 0x11, 0xE0, 0xA1, 0xB1, 0x1A, 0xE1 } },
+            { ".xlsx", new byte[] { 0x50, 0x4B, 0x03, 0x04 } },
+            { ".zip", new byte[] { 0x50, 0x4B, 0x03, 0x04, 0x50, 0x4B, 0x4C, 0x49, 0x54, 0x55, 0x50, 0x4B, 0x53, 0x70, 0x58, 0x50, 0x4B, 0x05, 0x06, 0x50, 0x4B, 0x07, 0x08, 0x57, 0x69, 0x6E, 0x5A, 0x69, 0x70 } },
+        };
+
+        // **WARNING!**
+        // In the following file processing methods, the file's content isn't scanned.
+        // In most production scenarios, an anti-virus/anti-malware scanner API is 
+        // used on the file before making the file available to users or other 
+        // systems. For more information, see the topic that accompanies this sample 
+        // app.
+
+        public static async Task<byte[]> ProcessFormFile<T>(IFormFile formFile,
+            ModelStateDictionary modelState, string[] permittedExtensions,
+            long sizeLimit)
+        {
+            var fieldDisplayName = string.Empty;
+
+            // Use reflection to obtain the display name for the model 
+            // property associated with this IFormFile. If a display
+            // name isn't found, error messages simply won't show
+            // a display name.
+            MemberInfo property =
+                typeof(T).GetProperty(
+                    formFile.Name.Substring(formFile.Name.IndexOf(".",
+                    StringComparison.Ordinal) + 1));
+
+            if (property != null)
+            {
+                if (property.GetCustomAttribute(typeof(DisplayAttribute)) is
+                    DisplayAttribute displayAttribute)
+                {
+                    fieldDisplayName = $"{displayAttribute.Name} ";
+                }
+            }
+
+            // Don't trust the file name sent by the client. Use Linq to
+            // remove invalid characters. Since the file name is displayed
+            // in error messages on the webform, the file name is also
+            // HTML encoded. Another option is to use
+            // Path.GetRandomFileName to generate a safe random
+            // file name.
+            var invalidFileNameChars = Path.GetInvalidFileNameChars();
+            var trustedFileName = WebUtility.HtmlEncode(
+                    invalidFileNameChars.Aggregate(
+                        formFile.FileName, (current, c) =>
+                            current.Replace(c, '_')));
+
+            // Check the file length. This check doesn't catch files that only have 
```

Comment:
What are you trying to do here? This has become more than a sample and started to veer into security sensitive guidance, which feels to me to be very out of scope for docs.

Reply:
We provided several of these guardrails when we put up the RP file uploads topic on https://github.com/aspnet/AspNetCore.Docs/pull/4265. You didn't object, so I assumed that you'd be ok with these approaches over here. Actually, I thought that you'd love the attention to security here. Sorry ... I should've asked you on the issue. The outline for these changes was approved, but only by the docs team.

Do you want me to strip all of the security of out the topic and sample app?

Equally, strip out the bit that I added to the other PR on custom model binding?

Follow-up on the other PR ... it has the dev uploading directly to the app's *wwwroot* folder. Do you want to leave that, too? If so, then that other PR (and the issue) can simply be closed.